tools/src/main/scala/scala/scalanative/codegen/llvm/compat/os/UnixCompat.scala

```diff
@@ -54,41 +49,21 @@ private[codegen] class UnixCompat(codegen: AbstractCodeGen)
     indent()
     line(s"$rec = $landingpad")
     line(s"$r0 = extractvalue $excRecTy $rec, 0")
-    line(s"$r1 = extractvalue $excRecTy $rec, 1")
-    line(s"$id = $typeid")
-    line(s"$cmp = icmp eq i32 $r1, $id")
-    line(s"br i1 $cmp, label %$excsucc, label %$excfail")
+    line(s"br label %$excsucc")
     unindent()
 
     line(s"$excsucc:")
     indent()
-    line(s"$w0 = call $ptrT $beginCatch($ptrT $r0)")
-    if (useOpaquePointers) {
-      line(s"$w2 = getelementptr ptr, ptr $w0, i32 1")
-      line(s"$exc = load ptr, ptr $w2")
-    } else {
-      line(s"$w1 = bitcast i8* $w0 to i8**")
-      line(s"$w2 = getelementptr i8*, i8** $w1, i32 1")
-      line(s"$exc = load i8*, i8** $w2")
-    }
-    line(s"call void $endCatch()")
-    str("br ")
+    line(s"$exc = call $ptrT $scalaNativeCatch($ptrT $r0)")
+    line("br ")
```

Comment:
Instead of upacking the ExceptionWrapper c++ class object, we simply call `scalanative_catch` which receives a `_UnwindException` and `return` the pointer to the Scala Native (java.lang.Throwable) exception object.
We assign `exc` to it and handle it.

Reply:
The only problem with this solution is fact that it might not be inlined, but we 'd need a benchmark to measure the impact in a code that is using exceptions heavily. We probably should try to inline it as it was done previously. Probably it would be easiest to implement if we'd get rid of `malloc`/`free` calls as well (mentioned above), but probably we can do it in the follow up PR.

Reply:
But the calls to `beginCatch` and `endCatch` were not inlined already, and those were to unpack the C++ exception object which we don't have now. It would be lovely to remove the `ExceptionWrapper` struct and add some reserved bytes inside `Throwable` (maybe a bunch of `Long`s) that could be used by the C code to store the `_Unwind_Exception`.
Since the exceptions are not in the hot path I would leave this to a follow-up PR.